video/vdu_stream_processor.h

```diff
@@ -27,7 +32,9 @@ class VDUStreamProcessor {
 		void wait_eZ80();
 		void sendModeInformation();
 	private:
-		Stream * stream;
+		std::shared_ptr<Stream> inputStream;
+		std::shared_ptr<Stream> outputStream;
+		std::shared_ptr<Stream> originalOutputStream;
```

Comment:
input and output have been split here, and the "originalOutputStream" is kept track of so that we can always reset our output to that

these have changed to use `shared_ptr` so they are safer - setting an output stream to a buffer that is subsequently erased would have otherwise left us with a pointer to a deleted object